packages/app/src/cli/services/generate/fetch-template-specifications.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  RemoteTemplateSpecification,
+  RemoteTemplateSpecificationsQuery,
+  RemoteTemplateSpecificationsQuerySchema,
+} from '../../api/graphql/template_specifications.js'
+import {TemplateSpecification} from '../../models/app/template.js'
+import {BaseFunctionConfigurationSchema, ZodSchemaType} from '../../models/extensions/schemas.js'
+import {ExtensionCategory} from '../../models/app/extensions.js'
+import {FunctionConfigType} from '../../models/extensions/functions.js'
+import {blocks} from '../../constants.js'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+
+export async function fetchTemplateSpecifications(token: string): Promise<TemplateSpecification[]> {
+  try {
+    const result: RemoteTemplateSpecificationsQuerySchema = await partnersRequest(
+      RemoteTemplateSpecificationsQuery,
+      token,
+    )
+    return result.templateSpecifications.map(mapRemoteTemplateSpecification)
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch (error) {
+    return []
+  }
+}
+
+export function mapRemoteTemplateSpecification(
+  remoteTemplateSpecification: RemoteTemplateSpecification,
+): TemplateSpecification {
+  return {
+    identifier: remoteTemplateSpecification.identifier,
+    name: remoteTemplateSpecification.name,
+    group: remoteTemplateSpecification.group,
+    supportLinks: remoteTemplateSpecification.supportLinks,
+    types: remoteTemplateSpecification.types.map((extension) => {
+      const extensionCustoms = resolveExtensionCustoms(extension.type)
+      return {
+        identifier: extension.type,
+        externalIdentifier: extension.type,
+        externalName: extension.type,
+        gated: false,
+        registrationLimit: registrationLimit(extension.type),
+        supportedFlavors: extension.supportedFlavors,
+        group: remoteTemplateSpecification.group,
+        category: () => extensionCustoms.category,
+        configSchema: BaseFunctionConfigurationSchema,
+        templateURL: remoteTemplateSpecification.url,
+        helpURL: remoteTemplateSpecification.supportLinks[0]!,
+        templatePath: (flavor: string) => {
+          const supportedFlavor = extension.supportedFlavors.find((supportedFlavor) => supportedFlavor.value === flavor)
+          if (!supportedFlavor) return undefined
+          return supportedFlavor.path
+        },
+      }
+    }),
+  }
+}
+
+// There should be another api with the extensions specifications. Right now templates only support functions
+function resolveExtensionCustoms(_type: string): {
+  category: ExtensionCategory
+  configSchema?: ZodSchemaType<FunctionConfigType>
+} {
+  return {
+    category: 'function',
+    configSchema: BaseFunctionConfigurationSchema,
+  }
+}
```

Comment:
why do you need this function? 🤔

Reply:
This are the missing things either in the template spec or the extensions. 

We need to cover these requirements. Now it's hardcoded to the functions but we should add[ this information ](https://github.com/Shopify/cli/pull/1541#discussion_r1140148549) to the current extensions spec

Reply:
You are not using the configSchema anywhere. But in any case and related to a previous comment, the schema doesn't belong to the template, schema is a "build" thing and the template is a "generate" thing.

Reply:
I mean the one in this function, the schema is used yes :P 
The schema can still be defined locally for now, that part is not relevant for the templates

Reply:
I've removed the schema from there.